packages/@lwc/engine-dom/src/custom-elements/create-scoped-registry.ts

```diff
@@ -19,6 +19,20 @@ export type CreateScopedConstructor = (
     UserCtor: CustomElementConstructor
 ) => CustomElementConstructor;
 
+type WhenDefinedCallback = (
+    ctor: CustomElementConstructor | PromiseLike<CustomElementConstructor>
+) => void;
+
+interface Definition {
+    UserCtor: CustomElementConstructor;
+    PivotCtor?: CustomElementConstructor;
+    connectedCallback?: () => void;
+    disconnectedCallback?: () => void;
+    adoptedCallback?: () => void;
+    attributeChangedCallback?: (name: string, oldValue: any, newValue: any) => void;
+    observedAttributes: Set<string>;
+}
```

Comment:
I just decided to move this up to have all the TypeScript types/interfaces together.